src/adapter/src/explain_new/mir/text.rs

```diff
@@ -192,15 +192,26 @@ impl<'a> Displayable<'a, MirRelationExpr> {
                             }
                             ctx.indent -= 1;
                         }
-                        JoinImplementation::IndexedFilter(_, key, row) => {
+                        JoinImplementation::IndexedFilter(_, key, rows) => {
                             write!(f, "{}Lookup ", ctx.indent)?;
 
-                            for (i, (expr, lit)) in std::iter::zip(key, row.unpack()).enumerate() {
-                                Displayable::from(expr).fmt_text(f, &mut ())?;
-                                write!(f, " = ")?;
-                                write!(f, "{}", lit)?;
-                                if i < key.len() - 1 {
-                                    write!(f, " AND ")?;
+                            debug_assert_eq!(inputs.len(), 2);
+
+                            for (i, row) in rows.iter().enumerate() {
+                                for (j, (expr, lit)) in
+                                    std::iter::zip(key, row.unpack()).enumerate()
+                                {
+                                    write!(f, "(")?;
+                                    Displayable::from(expr).fmt_text(f, &mut ())?;
+                                    write!(f, " = ")?;
+                                    write!(f, "{}", lit)?;
+                                    if j < key.len() - 1 {
+                                        write!(f, " AND ")?;
+                                    }
+                                    write!(f, ")")?;
+                                }
```

Comment:
I did it, but it turned out a bit weird, because in the middle of the 2 `separated` calls I had to use `write!` to create a new `Display` out of 2 `Display`s, and I could do this only by creating the `closure_to_display` helper method. I’m not sure if I maybe missed some more straightforward way due to my limited Rust experience.